### PR TITLE
Changing variorum_set_topology() to use "get".

### LINKED ISSUE #57: Topology interface names are confusing.
From `msr_core.c`

    int nthreads;
    variorum_set_topology(NULL, NULL, &nthreads);

This should be named `variorum_get_topology()`.  Using "set" establishes an expectation that we're limiting the topology to a subset of what's available.  

From `config_architecture.c`

    int variorum_get_topology(void)
    {
        ....
        hwloc_topology_init(&topology);
        hwloc_topology_load(topology);

This should be `variorium_initialize_topology()`.  Using "get" in isolation is fine, but there's a stronger case for using "get" elsewhere, and we really are doing an initialization pass here.

There's a third problem:  the existing "set" doesn't check to see if the existing "get" has run.  That's less of an issue as this API isn't exposed to users, but I'm planning on adding it to the expert side.  Since the initialization function shouldn't need to be called very often, I'm thinking it's best to combine the two functions and have a static initialization flag to check to see if initialization is required.  That would remove our ability to force a reinitization, but I don't think we have a use case that needs that feature.

### PR DESCRIPTION
**Pull request by rountree**
Fixes #57

Removed variorum_set_topology, as the name was confusing

Merged functionality of variorum_set_topology into variorum_get_topology.

variorum_get_topology now executes initialization on first use.

Changed return code to void and added assertions for error checking on the assumption that if the underlying hwloc calls are failing, there's no point in trying to continue and failure should be dramatic.

Updated all calls in variorum to account for the above.

### CODE REVIEW

File: src/variorum/config_architecture.c [hunk 1]
```diff
@@ -35,12 +36,7 @@ int variorum_enter(const char *filename, const char *func_name, int line_num)
 
     variorum_init_func_ptrs();
 
-    err = variorum_get_topology();
-    if (err)
-    {
-        variorum_error_handler("Cannot get topology", err, getenv("HOSTNAME"), __FILE__, __FUNCTION__, __LINE__);
-        return err;
-    }
+    variorum_get_topology(NULL, NULL, NULL); //Triggers initialization on first call.  Errors assert.
```
Comment (slabasan) [inline src/variorum/config_architecture.c]: Add space here: `// Triggers`

File: src/variorum/variorum.h [hunk 2]
```diff
@@ -138,8 +138,8 @@ int variorum_print_hyperthreading(void);
 
 /// @brief Print architecture topology in long format.
 ///
-/// @return Error code.
-int variorum_print_topology(void);
+/// @no return.
```
Comment (slabasan) [inline src/variorum/variorum.h]: Since it returns void, you can remove this line of the comment.

**Comment by rountree** [comment]
Ran astyle across all of the .h, .c and .cpp files in the project.  Lots of small formatting changes.  Not great combining them in this PR but simpler than separating out the files I touched with the code changes.

**Comment by slabasan** [comment]
I'm surprised that some of the files needed formatting, such as `powmon/power_wrapper_dynamic.c`. When I checkout `dev`, mine says no formatting changes. Can you double check your latest commits? Does it match what is at the tip of `dev`?

**Comment by rountree** [comment]
Killing this off to deal with formatting issues first.

**Pull request closed without merging**